.github/workflows/cicd.yml

```diff
@@ -1,91 +1,77 @@
+      - name: Set the environment based on the branch
+        id: define_environment
+        run: |
+          if [ "${{ github.ref }}" = "refs/heads/main" ]; then
+            echo "env_name=staging" >> $GITHUB_OUTPUT
+          elif [ "${{ github.ref }}" = "refs/heads/dev" ]; then
+            echo "env_name=development" >> $GITHUB_OUTPUT
+          elif [ "${{ github.ref }}" = "refs/heads/production" ]; then
+            echo "env_name=production" >> $GITHUB_OUTPUT
+          fi
+      - name: Print the environment
+        run: echo "The environment is ${{ steps.define_environment.outputs.env_name }}"
 
-      - name: Set up Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: "3.10"
+    outputs:
+      env_name: ${{ steps.define_environment.outputs.env_name }}
 
-      - name: Install Python dependencies
-        run: pip install black flake8
+  deploy:
+    name: Deploy to ${{ needs.define-environment.outputs.env_name }} 🚀
+    runs-on: ubuntu-latest
+    if: ${{ needs.define-environment.outputs.env_name }}
+    needs: [gitflow-enforcer, define-environment]
+    environment: ${{ needs.define-environment.outputs.env_name }}
+    concurrency: ${{ needs.define-environment.outputs.env_name }}
 
-      - name: Run linters
-        uses: wearerequired/lint-action@v2
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
         with:
-          continue_on_error: false
-          black: true
-          flake8: true
-          flake8_args: "--ignore E1,E2,E3,E5,W1,W2,W3,W5" # black already handles formatting, this prevents conflicts
-
-  deploy-to-dev:
-    needs: run-linters
-    if: github.ref_name == 'dev'
-    concurrency: development
-    uses: "./.github/workflows/deploy.yml"
-    with:
-      environment: development
-      env-file: ".env_dev"
-      stage: "dev"
-      role-session-name: "veda-data-airflow-github-development-deployment"
-      aws-region: "us-west-2"
-
-    secrets: inherit
-
-  deploy-to-staging:
-    needs: run-linters
-    if: github.ref_name == 'main'
-    concurrency: staging
-    uses: "./.github/workflows/deploy.yml"
-    with:
-      environment: staging
-      env-file: ".env_staging"
-      stage: "staging"
-      role-session-name: "veda-data-airflow-github-staging-deployment"
-      aws-region: "us-west-2"
-    secrets: inherit
-
-  deploy-to-production:
-    needs: run-linters
-    if: github.ref_name == 'production'
-    concurrency: production
-    uses: "./.github/workflows/deploy.yml"
-    with:
-      environment: production
-      env-file: ".env_prod"
-      stage: "production"
-      role-session-name: "veda-data-airflow-github-production-deployment"
-      aws-region: "us-west-2"
+          lfs: "true"
+          submodules: "recursive"
+        
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ secrets.DEPLOYMENT_ROLE_ARN }}
```

Comment:
Do we lose our ability to deploy to UAH here where we do not have a deployment role? Or have we never had github push actions for cycling changes to the dev and staging MWAAs in UAH?

Reply:
yep, the current workflow https://github.com/NASA-IMPACT/veda-data-airflow/blob/dev/.github/workflows/deploy.yml already uses it